### ISSUE TITLE
Формирование правильного сообщения

### ISSUE THREAD

**Issue by Morfey216** [Vadimfre/keksagram-Semashko#16]
Мне кажется, что на этом скриншоте не только одна указанная ошибка...
![image](https://github.com/Vadimfre/keksagram-Semashko/assets/42171361/eb0d0b41-cf11-4570-9527-d3e501f91dbb)
Было бы правильным формировать сразу полное сообщение, в котором перечислить все ошибки, чтобы пользователь сразу их исправил

**Comment by Vadimfre** [comment]
https://github.com/Vadimfre/keksagram-Semashko/blob/252a3911963fd2b065bb99414eec3e8019e19d8e/main.js#L17  вот так вот решил тут и пример и все понятно вроде

**Comment by Morfey216** [comment]
> https://github.com/Vadimfre/keksagram-Semashko/blob/252a3911963fd2b065bb99414eec3e8019e19d8e/main.js#L17
> 
> вот так вот решил тут и пример и все понятно вроде

Я имел ввиду то, что в сообщении нужно указать сразу все особенности, все имеющиеся ошибки, а не одну подробно

**Comment by Vadimfre** [comment]
да, это удобнее  https://github.com/Vadimfre/keksagram-Semashko/blob/73075e283ab2f4e9a543180b942678706856c973/main.js#L17-L24

**Comment by Morfey216** [comment]
То есть ты думаешь, что так правильнее?

![image](https://github.com/Vadimfre/keksagram-Semashko/assets/42171361/89d27b28-117a-49a7-9d6c-cbd73e515056)

**Comment by Vadimfre** [comment]
а вот тут что-то я не понял, не могли бы пояснить? Просто с самого начала выводить на экран это большое сообщение? А если пользователь допускает ошибку, то говорить ему что за ошибка конкретно?

**Comment by Morfey216** [comment]
Ну давай смотреть пример

![image](https://github.com/Vadimfre/keksagram-Semashko/assets/42171361/dcb84a4b-e0e9-4b95-af66-845b2f2c4996)

Что именно из перечисленного нарушено? Как обычному пользователю догадаться, чего хочет от него программа?

Правильное решение будет заключаться в таком алгоритме:
Есть список проверок на валидность. Каждому пункту проверки должно быть поставлено в соответствие отдельное описание ошибки. То есть если у тебя по ТЗ было бы всего 2 проверки, то и описания тоже два, а если 22 проверки, то и 22 описания. Один к одному.
В нашем ТЗ всего 6 различных проверок и все они прописаны
![image](https://github.com/Vadimfre/keksagram-Semashko/assets/42171361/c1de34bd-0014-450a-b047-00b026c98996)
Соответственно строим структуру данных для обработки ошибок
А) в перечислении (про которое нейминг фиксили) должно быть 6 пунктов вида тип ошибки: текст сообщения
Б) создаем еще одну структуру для учета состояния ошибок. Вид структуры - тип ошибки: состояние (есть/нету). Первоначально все типы ошибок выключены.
В) перечисление с этого момента перестанет быть перечислением и станет ассоциативным словарем. Нейминг строится уже иначе, но об этом позже.

Проверка: проходишь циклом по каждому хэштегу и проверяешь его на ошибку. Если ошибка есть - включаешь ее индикатор в объекте учета состояния. 
Перед отправкой формы динамически формируешь строку-сообщение об ошибках - проверяешь состояние ошибки в объекте учета состояний и если ошибка есть - идешь в словарь за текстом сообщения для этого пункта. Текст добавляется к строке для вывода. Затем проверяешь состояние следующей ошибки и тд. И только потом выводишь получившееся сообщение

**Comment by Vadimfre** [comment]
Спасибо, понял, разобрался. сначала создал вот такой словарь
 https://github.com/Vadimfre/keksagram-Semashko/blob/d2ec025a27a3c3fd9d4270f4493993185008953a/main.js#L23C1-L30 
 дальше добавляю обработчик на кнопку, он удаляется  в функции hideImageEditForm https://github.com/Vadimfre/keksagram-Semashko/blob/d2ec025a27a3c3fd9d4270f4493993185008953a/main.js#L288 
 вызывается вот функция, которая сначала забирает хеш-теги, проверяет их и возвращает уже готовое сообщение об ошибке, дальше просто вставляю это сообщение, чтобы пользователь увидел ошибки, либо отправил форму
  https://github.com/Vadimfre/keksagram-Semashko/blob/ac23ceb78511574fb626fbdf5697bd9a87a9dce0/main.js#L268-L276
  Функция  validateHashtags теперь работает по другому, если же она находит ошибку то помечает ее в словаре errorsState
  https://github.com/Vadimfre/keksagram-Semashko/blob/ac23ceb78511574fb626fbdf5697bd9a87a9dce0/main.js#L284C2-L309
  Еще в коде появилась новая функция которая сбрасывает errorsState https://github.com/Vadimfre/keksagram-Semashko/blob/ac23ceb78511574fb626fbdf5697bd9a87a9dce0/main.js#L324-L328 
  и вот функция которая генерирует сообщение https://github.com/Vadimfre/keksagram-Semashko/blob/ac23ceb78511574fb626fbdf5697bd9a87a9dce0/main.js#L335-L343

**Comment by Vadimfre** [comment]
но тут есть еще одна проблемка, если написать #ggg## то все равно отправит, а вот мне такого не нужно, должна выводится ошибка: 'Хэштеги должны разделяться пробелами'
И вот как исправить я вообще не понимаю, не могли бы помочь?:)
